core.js

```diff
@@ -1,79 +1,133 @@
+      }
+      if (toIgnore) continue;
+
+      // Handle recursive paths
+      if (fs.lstatSync(dir + path.sep + site).isDirectory()) {
+        pathMap = {
+          ...pathMap,
+          ...this.buildPathMap(dir + path.sep + site)
+        };
+
+        continue;
+      }
 
+      // Is file
+      let fileExtension = site.split(".").pop().length;
+      let fileNameWithoutExtension = site.substring(
+        0,
+        site.length - (fileExtension + 1)
+      );
+      fileNameWithoutExtension =
+        this.ignoreIndexFiles && fileNameWithoutExtension === "index"
+          ? ""
+          : fileNameWithoutExtension;
+      let newDir = dir.replace(this.pagesdirectory, "").replace(/\\/g, "/");
+
+      let pagePath = newDir + "/" + fileNameWithoutExtension;
+      pathMap[pagePath] = {
+        page: pagePath
+      };
     }
 
-    preLaunch() {
-        fs.writeFileSync(`${this.targetDirectory}sitemap.xml`, this.sitemap, {flag: 'w'});
-    };
-
-    finish() {
-        fs.writeFileSync(`${this.targetDirectory}sitemap.xml`, '</urlset>', {flag: 'as'});
-
-    };
-
-    sitemapMapper(dir) {
-
-        let date = dateFns.format(new Date(), 'YYYY-MM-DD');
-        let data = fs.readdirSync(dir);
-        for (let site of data) {
-            if (site[0] === '_' || site[0] === '.') continue;
-            let toIgnore = false;
-            for (let path of this.ignoredPaths) {
-                if (site.includes(path)) toIgnore = true;
-            }
-            if (toIgnore) continue;
-
-            if (fs.lstatSync(dir + path.sep + site).isDirectory()) {
-                this.sitemapMapper(dir + path.sep + site);
-                continue;
-            }
-            let fileExtension = site.split('.').pop().length;
-            let fileNameWithoutExtension = site.substring(0, site.length - (fileExtension + 1));
-            fileNameWithoutExtension = this.ignoreIndexFiles && fileNameWithoutExtension === 'index' ? '' : fileNameWithoutExtension;
-            let newDir = dir.replace(this.pagesdirectory, '').replace(/\\/g, '/');
-            let alternates = '';
-            for (let langSite in this.alternatesUrls) {
-                alternates += `<xhtml:link rel="alernate" hreflang="${langSite}" href="${
-                    this.alternatesUrls[langSite]
-                    }${newDir + '/' + fileNameWithoutExtension}" />`;
-            }
-            let xmlObject = `<url><loc>${this.baseUrl}${newDir +
-                '/' +
-                fileNameWithoutExtension}</loc>` +
-                alternates +
-                `<lastmod>${date}</lastmod></url>`;
-            fs.writeFileSync(`${this.targetDirectory}${path.sep}sitemap.xml`, xmlObject, {flag: 'as'});
-        }
+    return pathMap;
+  }
+
+  async sitemapMapper(dir) {
+    var pathMap = this.buildPathMap(dir);
+    const exportPathMap = this.nextConfig && this.nextConfig.exportPathMap;
+
+    if (exportPathMap) {
+      pathMap = await exportPathMap(pathMap, {});
```

Comment:
Hi, awesome work.

If export map is true is a boleean and you using it as function here.
If I wrong sorry, but can you explain that.